app/src/experiments.py

```diff
@@ -0,0 +1,153 @@
+import argparse
+import json
+import logging
+from pprint import pformat
+from typing import Any, Dict
+
+from phoenix.client import Client
+from phoenix.client.experiments import run_experiment
+from phoenix.client.resources.datasets import Dataset
+from phoenix.client.resources.experiments.types import TaskOutput
+
+from src.common import phoenix_utils
+from src.pipelines.generate_referrals.pipeline_wrapper import PipelineWrapper
+
+logger = logging.getLogger(__name__)
+
+
+def get_sets(output: TaskOutput, expected: Dict[str, Any]) -> tuple[set[str], set[str]]:
+    assert isinstance(output, dict), f"Expected dict, but got {type(output)}: {pformat(output)}"
+    assert len(output["content"]) == 1, pformat(output["content"])
+    output_obj = json.loads(output["content"][0]["text"])
+    # Extract only the names of the resources from the output
+    output_set = set([resource["name"] for resource in output_obj["resources"]])
+
+    expected_json_str = expected[OUTPUT_COLUMN_NAME]
+    expected_obj = json.loads(expected_json_str)
+    expectation_set = set(expected_obj["expected_referrals"])
+
+    return output_set, expectation_set
+
+
+def precision(output: TaskOutput, expected: Dict[str, Any]) -> float:
+    output_set, expectation_set = get_sets(output, expected)
+    return len(output_set.intersection(expectation_set)) / len(expectation_set)
+
+
+def accuracy(output: TaskOutput, expected: Dict[str, Any]) -> float:
+    output_set, expectation_set = get_sets(output, expected)
+    return len(output_set.intersection(expectation_set)) / len(output_set)
+
+
+pipeline_wrapper = PipelineWrapper()
+pipeline_wrapper.setup()
+
+
+def get_answer(example: dict) -> TaskOutput:
+    question_json_str = example["input"][INPUT_COLUMN_NAME]
+    question_obj = json.loads(question_json_str)
+    question = question_obj["caseworker_input"]
+
+    logger.info("Getting answer for: %r", question)
+    response = pipeline_wrapper.run_api(query=question)
```

Comment:
This runs the pipeline in the local code against the local DB.
We may want an option to run against the deployed application, in which case the URL would need to be provided.